src/ui/ui_nbgl.c

```diff
@@ -172,40 +172,61 @@ void ui_display_public_key_flow(void) {
                             display_addr, address_verification_cancelled);
 }
 
-
 //  ----------------------------------------------------------- 
-//  ---------------- SIGN TRANSACTION FLOW --------------------
+//  ---------------- SIGN FLOW COMMON -------------------------
 //  ----------------------------------------------------------- 
-
-#define MAX_TAG_VALUE_PAIRS_DISPLAYED (3)
-static nbgl_layoutTagValue_t pairs[MAX_TAG_VALUE_PAIRS_DISPLAYED];
-static nbgl_layoutTagValueList_t pair_list = {0};
-static nbgl_pageInfoLongPress_t info_long_press;
-static const char *warning_msg;
+static signType_t sign_type = SIGN_TRANSACTION;
 
 static void ui_display_action_sign_done(bool validated) 
 {
     if(validated) 
     {
-        nbgl_useCaseStatus("TRANSACTION\nSIGNED", true, ui_idle);
+        if(sign_type == SIGN_TRANSACTION){
+            nbgl_useCaseStatus("TRANSACTION\nSIGNED", true, ui_idle);
+        }else if(sign_type == SIGN_MESSAGE){
+            nbgl_useCaseStatus("MESSAGE\nSIGNED", true, ui_idle);
+        }else{
+            nbgl_useCaseStatus("CERTIFICATE\nSIGNED", true, ui_idle);
+        }
     }
     else 
     {
-        nbgl_useCaseStatus("Transaction rejected", false, ui_idle);
+        if(sign_type == SIGN_TRANSACTION){
+            nbgl_useCaseStatus("Transaction rejected", false, ui_idle);
+        }else if(sign_type == SIGN_MESSAGE){
+            nbgl_useCaseStatus("Message rejected", false, ui_idle);
+        }else{
+            nbgl_useCaseStatus("Certificate rejected", false, ui_idle);
+        }
     }
 }
 
-static void transaction_rejected(void) {
+static void sign_rejected(void) {
     io_seproxyhal_touch_cancel();
     // Display "rejected" screen
     ui_display_action_sign_done(false);
 }
 
 static void reject_confirmation(void) {
-    nbgl_useCaseConfirm("Reject transaction?", NULL, "Yes, Reject", "Go back to transaction", transaction_rejected);
+    if(sign_type == SIGN_TRANSACTION){
+        nbgl_useCaseConfirm("Reject transaction?", NULL, "Yes, Reject", "Go back to transaction", sign_rejected);
+    }else if(sign_type == SIGN_MESSAGE){
+        nbgl_useCaseConfirm("Reject message?", NULL, "Yes, Reject", "Go back to message", sign_rejected);
+    }else{
+        nbgl_useCaseConfirm("Reject certificate?", NULL, "Yes, Reject", "Go back to certificate", sign_rejected);
+    }
 }
```

Comment:
same comment here.

Reply:
How about the go-back hint, `Go back to transaction` or `Go back to message` ?

Reply:
same for this button, should be also `transaction`